### ISSUE TITLE
getSubscriptions returns empty array on IOS (everything working fine on Android)

### ISSUE THREAD

**Issue by chrisdev0342** [hyochan/react-native-iap#2208]
**Description**
I am fetching renewable subscriptions in IOS, which always returns an empty array. Although everything is working fine on Android. I also tried to run `requestSubscription` with one of the `sku` but it throws an error:

> E_DEVELOPER_ERROR Invalid product ID

I am trying to fetch like:
`const subscription = await RNIap.getSubscriptions({ skus: ["subscription_01", "subscription_01"] });`

I've been struggling with this from past week, I have tried to push build to TestFlight and test but no success. Any help will be much appreciated. 

**Expected Behavior**
Ideally it should return an array of subscriptions just like Android.

**Environment:**

- react-native-iap: 12.4.8
- react-native: 0.70.5
- Platforms (iOS, simulator, device):

**Comment by andresesfm** [comment]
Is this on a device with iOS 15 or newer? have you tried using Storekit2 mode?

**Comment by chrisdev0342** [comment]
> Is this on a device with iOS 15 or newer? have you tried using Storekit2 mode?

Yes, using StoreKit it works in debug mode but after deploying on TestFlight does not work.

**Comment by andresesfm** [comment]
I have seen this on other tickets. It might be the kind of subscription. For iOS 15+, using storekit2 mode seems to solve it.

**Comment by 5ylvino** [comment]
I am currently facing the same issue with subscriptions fetching an empty array. But let me try the Storekit2 mode and see how it goes.

**Comment by 5ylvino** [comment]
Wow! I woke up this morning, I did the Storekit configuration. And everything started working perfectly as supposed. Though, I ensured that everything matched the values I have on the apple console.

**Comment by oloyedewaris** [comment]
I've been facing this same issue of fetching empty array for about 2 weeks now



Please how do I use storekit2?

**Comment by 5ylvino** [comment]
@oloyedewaris follow the step in this documentation ===> https://www.revenuecat.com/blog/engineering/ios-in-app-subscription-tutorial-with-storekit-2-and-swift/#h-setting-up-app-store-connect

**Comment by itsdenispavlovic** [comment]
I am getting the same error, but on purchase. I am using expo 49

**Comment by sanjaysoalnki2000** [comment]
> > Is this on a device with iOS 15 or newer? have you tried using Storekit2 mode?
> 
> Yes, using StoreKit it works in debug mode but after deploying on TestFlight does not work.

i facing same problem.

**Comment by inayatAliDev** [comment]
I am also getting same issue

all steps done (storekit confiuguration) as described here:
https://developer.apple.com/documentation/xcode/setting-up-storekit-testing-in-xcode#Enable-StoreKit-Testing-in-Xcode

but now getting empty array when try to fetch subscritions

**Comment by anwargul0x** [comment]
any updates on this issue?

**Comment by Jova266** [comment]
Commenting so this thread stays relevant!

**Comment by devburiti** [comment]
Facing the same problem ):

**Comment by yairvillarp** [comment]
same problem here as well

**Comment by Jova266** [comment]
I solved the problem when I completed all the neccessery things in the Business tab in appstoreconnect. Tax fomrs, Bank account and Paid app agreement.

**Comment by inayatAliDev** [comment]
> I am also getting same issue
> 
> all steps done (storekit confiuguration) as described here: https://developer.apple.com/documentation/xcode/setting-up-storekit-testing-in-xcode#Enable-StoreKit-Testing-in-Xcode
> 
> but now getting empty array when try to fetch subscritions

Solved after adding bank information which was necessary for paid agreement

**Comment by AkramBadah** [comment]
Do I need storekit confiuguration for test flight?

**Comment by samya-djenad** [comment]
Did someone find a solution ?

**Comment by hyochan** [comment]
I’m closing all issues reported in versions below 14, as the library now supports the new architecture with NitroModules and has been completely revamped.

FYI, `getSubscriptions` is no longer available and this is replaced with `fetchProducts` with type `subs`.
